client/src/Api.js

```diff
@@ -94,15 +94,32 @@ const Api = {
       });
     },
     histogram({
-      path, field,
-      contest_id, is_tester,
-      start_date, end_date,
-      bin_size, bin_count, bin_custom,
+      path,
+      field,
+      contest_id,
+      is_tester,
+      start_date,
+      end_date,
+      bin_size,
+      bin_count,
+      bin_custom,
     }) {
+      if (contest_id) {
+        [start_date, end_date] = "";
+      }
```

Comment:
Nice! 

Destructuring from an empty string exploits one of the ugliest "features" of javascript, accessing out-of-bounds indices from iterables, and can often be seen as a code smell.

Not required (and you can leave this as-is), 
but a good alternative is self-documenting variable names, that make it very clear that the mutual exclusivity is because the are related to the dates. Contests already define a start and end date, which could be helpful for the future reader/maintainer of the code:

```suggestion
      const exclusiveDateParams = contest_id ? { contest_id } : { start_date, end_date };
```

```js
{
params: {
   ...exclusiveDateParams,
   is_tester,
   field,
   // etc
}

```